File: content/problems/2831-find-the-longest-equal-subarray/solutions/position-lists-sliding-window/solution.cpp
class Solution {
    public:int longestEqualSubarray(vector<int>& nums,int k){
        unordered_map<int,vector<int>> positions;
        for(int i=0;i<(int)nums.size();++i)positions[nums[i]].push_back(i);
        int answer=0;
        for(auto& [value,list]:positions){
            int left=0;
            for(int right=0;right<(int)list.size();++right){
                while(list[right]-list[left]-(right-left)>k)++left;
                answer=max(answer,right-left+1);
            }
        }
        return answer;
    }
};

File: content/problems/2831-find-the-longest-equal-subarray/solutions/position-lists-sliding-window/Solution.java
class Solution {
    public int longestEqualSubarray(List<Integer> nums,int k){
        Map<Integer,List<Integer>> positions=new HashMap<>();
        for(int i=0;i<nums.size();i++)positions.computeIfAbsent(nums.get(i),key->new ArrayList<>()).add(i);
        int answer=0;
        for(List<Integer> list:positions.values()){
            int left=0;
            for(int right=0;right<list.size();right++){
                while(list.get(right)-list.get(left)-(right-left)>k)left++;
                answer=Math.max(answer,right-left+1);
            }
        }
        return answer;
    }
}
